client-js/app/elements/labrad-plot.ts

```diff
@@ -470,206 +787,186 @@ export class Plot extends polymer.Base {
-                .attr('width', w)
-                .attr('height', h)
-                .style('fill', (d) => getColor(d[p.displaySurface], zMin, zMax));
-        }
-        break;
+    switch (this.drawMode2D) {
+    case 'dots':
+      wScreen = this.dotBaseSize;
+      hScreen = wScreen;
+      break;
 
-      case 'vargrid':
-        for (let row of data) {
-          this.chartBody
-                .append('rect')
-                .datum(row)
-                .classed('data', true)
-                .attr('x', (d) => p.xScale(d[0]))
-                .attr('y', (d) => p.yScale(p.yNext[d[1]]))
-                .attr('width', (d) => p.xScale(p.xNext[d[0]]) - p.xScale(d[0]))
-                .attr('height', (d) => {
-                  return Math.abs(p.yScale(p.yNext[d[1]]) - p.yScale(d[1]));
-                })
-                .style('fill', (d) => getColor(d[p.displaySurface], zMin, zMax));
-        }
-        break;
+    case 'rectfill':
+      wScreen = Math.abs(this.xScale(this.dx0) - this.xScale(0));
+      hScreen = Math.abs(this.yScale(this.dy0) - this.yScale(0));
+      break;
 
-      default:
-        // Nothing to do.
-        break;
+    case 'vargrid':
+      wScreen = this.xScale(this.xNext[x]) - this.xScale(x),
+      hScreen = Math.abs(this.yScale(this.yNext[y]) - this.yScale(y));
+      y = this.yNext[y];
+      break;
     }
-  }
-
-
-  private installMouseListeners() {
-    if (!this.svg) return;
-    switch (this.mouseMode) {
-      case 'pan':
-        this.svg.on('mousedown', null);
-        this.svg.call(this.zoom);
-        this.$.plot.style.cursor = 'auto';
-        this.$.plot.style.cursor = 'all-scroll';
-        break;
 
-      case 'zoomRect':
-        this.svg.on('.zoom', null)
-        this.svg.on('mousedown', () => this.zoomRectangle());
-        this.$.plot.style.cursor = 'auto';
-        this.$.plot.style.cursor = 'crosshair';
-        break;
+    const xScreen = this.xScale(x) + (wScreen / 2);
+    const yScreen = this.yScale(y) + (hScreen / 2);
 
-      default:
-        // Nothing to do.
-        break;
-    }
+    outputObject.x = xScreen;
+    outputObject.y = yScreen;
+    outputObject.w = wScreen;
+    outputObject.h = hScreen;
   }
 
 
-  private updatePlotStyles() {
-    if (!this.svg) return;
-    switch (this.numIndeps) {
-      case 1:
-        this.$$('rect.background').style.fill = '#ffffff';
-        this.$.modes2d.style.visibility = 'hidden';
-        this.is1D = true;
-        this.is2D = false;
-        break;
-
-      case 2:
-        this.$$('rect.background').style.fill = '#222222';
-        this.is1D = false;
-        this.is2D = true;
-        break;
-
-      default:
-        // Nothing to do.
-        break;
-    }
+  /**
+   * Projects a coordinate in screen coordinates (relative to the canvas) to
+   * world coordinates (relative to the 3D projection of the scene).
+   */
+  private projectScreenCoordToWorldCoord(x: number, y: number, outputVector: THREE.Vector3) {
```

Comment:
I would reorder to put this after `projectGraphPositions`, which describes the various coordinate systems.